enterprise/tailnet/pgcoord_test.go

```diff
@@ -693,8 +693,79 @@ func assertEventuallyHasDERPs(ctx context.Context, t *testing.T, c *testConn, ex
 				t.Logf("expected DERP %d to be in %v", e, derps)
 				continue
 			}
+			return
+		}
+	}
+}
+
+func assertNeverHasDERPs(ctx context.Context, t *testing.T, c *testConn, expected ...int) {
+	t.Helper()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case nodes := <-c.nodeChan:
+			derps := make([]int, 0, len(nodes))
+			for _, n := range nodes {
+				derps = append(derps, n.PreferredDERP)
+			}
+			for _, e := range expected {
+				if slices.Contains(derps, e) {
+					t.Fatalf("expected not to get DERP %d, but received it", e)
```

Comment:
you call this from goroutines, which means you shouldn't use `Fatalf` as it doesn't stop the main test goroutine.

Maybe we should rename these functions to `requireXxxx` and `assertXxxx` to follow the testify convention about whether or not they're safe outside the main test goroutine.

Reply:
They're not goroutines, just inline functions so the defer works.